tgui/packages/tgui/interfaces/PersonalCrafting.tsx

```diff
@@ -1,311 +1,922 @@
+      <Stack.Item width="14px" textAlign="center">
+        <Icon name={TYPE_ICONS[type] || 'circle'} />
+      </Stack.Item>
+      <Stack.Item grow style={{ 'text-transform': 'capitalize' }}>
+        {type.toLowerCase()}
+      </Stack.Item>
+      <Stack.Item>
+        {type === 'Can Make' ? (
+          craftableCount
+        ) : diet.liked_food.includes(type) ? (
+          <Icon name="face-laugh-beam" color={'good'} />
+        ) : diet.disliked_food.includes(type) ? (
+          <Icon name="face-tired" color={'average'} />
+        ) : (
+          diet.toxic_food.includes(type) && (
+            <Icon name="skull-crossbones" color={'bad'} />
+          )
         )}
-        <div className="PersonalCraftingGridItem__craftability">
-          {recipe.craftable ? 'Craft' : 'Uncraftable'}
-        </div>
-      </div>
-    </div>
-  )) as any;
+      </Stack.Item>
+    </Stack>
+  );
 };
 
-const CompactCraftingList = (props: CraftingListProps, context) => {
-  const { recipes = [] } = props;
-  const { act } = useBackend<RawData>(context);
+const RecipeContentCompact = ({ item, craftable, busy, mode }, context) => {
+  const { act, data } = useBackend<Data>(context);
+  return (
+    <Section>
+      <Stack my={-0.75}>
+        <Stack.Item>
+          <Box
+            className={classes([
+              mode ? 'cooking32x32' : 'crafting32x32',
+              'a' + item.result,
+            ])}
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Stack>
+            <Stack.Item grow>
+              <Box mb={0.5} bold style={{ 'text-transform': 'capitalize' }}>
+                {item.name}
+              </Box>
+              <Box style={{ 'text-transform': 'capitalize' }} color={'gray'}>
+                {Array.from(
+                  Object.keys(item.reqs).map((atom_id) => {
+                    const name = data.atom_data[(atom_id as any) - 1]?.name;
+                    const is_reagent =
+                      data.atom_data[(atom_id as any) - 1]?.is_reagent;
+                    const amount = item.reqs[atom_id];
+                    return is_reagent
+                      ? name + '\xa0' + amount + 'u'
+                      : amount > 1
+                        ? name + '\xa0' + amount + 'x'
+                        : name;
+                  })
+                ).join(', ')}
+
+                {item.chem_catalysts &&
+                  ', ' +
+                    Object.keys(item.chem_catalysts)
+                      .map((atom_id) => {
+                        const name = data.atom_data[(atom_id as any) - 1]?.name;
+                        const is_reagent =
+                          data.atom_data[(atom_id as any) - 1]?.is_reagent;
+                        const amount = item.chem_catalysts[atom_id];
+                        return is_reagent
+                          ? name + '\xa0' + amount + 'u'
+                          : amount > 1
+                            ? name + '\xa0' + amount + 'x'
```

Comment:
what's with this weird unicode?

Reply:
just use \&nbsp; or just insert a space, what's the big deal?

Reply:
Space won't work because I need the name and amount be on the same line. In compact mode, all ingredients are listed in a row. &nbsp; didn't work when I tried. I was rendered as "&nbsp;" text. Do you know how to insert it there?

Reply:
&nsbp; is the html code for a space

Reply:
It's no-break space that makes glued parts stay on one line. Space doesn't work as a glue. Anyway, I don't know how to enter &nbsp; here to make it work, and `\xa0` is unicode for &nbsp;

Reply:
Non issue, `\xa0` works fine for this, `&nbsp;` is an html entity and works as such only when it appears in html / jsx syntax. If it appears inside a string, it doesn't get special treatment.

Reply:
Alternative syntax, if that makes it more readable.

```js
`${name}\xa0${amount}x`
```